autoload/neomake.vim

```diff
@@ -661,6 +708,31 @@ function! s:RegisterJobOutput(jobinfo, lines) abort
     endif
 endfunction
 
+function! s:GetChannelId(channel) abort
+    return ch_info(a:channel)['id']
+endfunction
+
+function! neomake#MakeHandlerVimStdout(channel, output) abort
+    call neomake#utils#DebugMessage('MakeHandlerVim: stdout: ' . a:channel)
+    let channel_id = s:GetChannelId(a:channel)
+    let jobinfo = s:jobs[channel_id]
+    call neomake#MakeHandler(channel_id, split(a:output, "\n", 1), 'stdout')
+endfunction
+
+function! neomake#MakeHandlerVimStderr(channel, output) abort
+    call neomake#utils#DebugMessage('MakeHandlerVim: stderr: ' . a:channel)
+    let channel_id = s:GetChannelId(a:channel)
+    let jobinfo = s:jobs[channel_id]
+    call neomake#MakeHandler(channel_id, split(a:output, "\n", 1), 'stderr')
+endfunction
+
+function! neomake#MakeHandlerVimClose(channel) abort
+    call neomake#utils#DebugMessage('Channel has been closed: ' . a:channel)
+    let channel_id = s:GetChannelId(a:channel)
+    let jobinfo = s:jobs[channel_id]
+    call neomake#MakeHandler(channel_id, job_info(jobinfo.job)['exitval'], 'exit')
+endfunction
```

Comment:
What do you have in mind for refactoring? I see this fine for now, as it's clearly handling Vim channel's callbacks, and acting as proxy to call MakeHandler.

I agree that the code (not just this Vim async part) should be refactored, I mean, the complete autoload/neomake.vim should be re designed / refactored... we can tackle this as an important step in another PR, but I'm all ears.